src/app/features/tech-record/components/change-vehicle-type/change-vehicle-type.component.ts

```diff
@@ -0,0 +1,99 @@
+import { Component, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { Location } from '@angular/common';
+import { GlobalErrorService } from '@core/components/global-error/global-error.service';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { CustomFormGroup, FormNode, FormNodeTypes } from '@forms/services/dynamic-form.types';
+import { TechRecordModel, VehicleTechRecordModel, VehicleTypes } from '@models/vehicle-tech-record.model';
+import { TechnicalRecordService } from '@services/technical-record/technical-record.service';
+import { take } from 'rxjs';
+import { getOptionsFromEnum } from '@forms/utils/enum-map';
+import { MultiOptions } from '@forms/models/options.model';
+
+@Component({
+  selector: 'app-change-vehicle-type',
+  templateUrl: './change-vehicle-type.component.html',
+  styleUrls: ['./change-vehicle-type.component.scss']
+})
+export class ChangeVehicleTypeComponent implements OnInit, OnChanges {
+  constructor(
+    public dfs: DynamicFormService,
+    public globalErrorService: GlobalErrorService,
+    private technicalRecordService: TechnicalRecordService,
+    private location: Location
+  ) {
+    this.technicalRecordService.selectedVehicleTechRecord$.pipe(take(1)).subscribe(data => (this.vehicleTechRecord = data));
+    this.technicalRecordService.editableTechRecord$.pipe(take(1)).subscribe(data => (this.currentTechRecord = data));
+  }
+
+  public currentTechRecord?: TechRecordModel;
+  public vehicleTechRecord?: VehicleTechRecordModel;
+  public form!: CustomFormGroup;
+
+  public template: FormNode = {
+    name: 'criteria',
+    type: FormNodeTypes.GROUP,
+    children: [
+      {
+        name: 'selectVehicleType',
+        label: 'Select a new vehicle type',
+        value: '',
+        type: FormNodeTypes.CONTROL
+      }
+    ]
+  };
+
+  ngOnChanges(changes: SimpleChanges): void {
+    this.globalErrorService.clearErrors();
+
+    const { vehicleTechRecord } = changes;
+
+    if (this.form && vehicleTechRecord?.currentValue && vehicleTechRecord.currentValue !== vehicleTechRecord.previousValue) {
+      this.form.patchValue(vehicleTechRecord.currentValue, { emitEvent: false });
+    }
+  }
+
+  get currentVrm(): string | undefined {
+    return this.vehicleTechRecord?.vrms.find(vrm => vrm.isPrimary === true)?.vrm;
+  }
+
+  get vehicleTypeOptions(): MultiOptions {
+    return getOptionsFromEnum(VehicleTypes);
+  }
+
+  ngOnInit(): void {
+    this.form = this.dfs.createForm(this.template) as CustomFormGroup;
+    if (this.vehicleTechRecord === undefined) {
+      this.navigateBack();
+    }
+  }
+
+  handleSubmitNewVehicleType(selectedVehicleType: string): void {
+    console.log(this.currentTechRecord);
```

Comment:
need to go

Reply:
This got removed when Tom pushed his changes, it was just to facilitate what he was doing